addons/im_livechat/static/src/legacy/public_livechat.js

```diff
@@ -233,10 +233,20 @@ var LivechatButton = Widget.extend({
         }
         def.then(function (livechatData) {
             if (!livechatData || !livechatData.operator_pid) {
-                self.displayNotification({
-                    message: _t("No available collaborator, please try again later."),
-                    sticky: true
-                });
+                try {
+                    self.displayNotification({
+                        message: _t("No available collaborator, please try again later."),
+                        sticky: true
+                    });
+                } catch (err) {
+                    /**
+                     * Failure in displaying notification happens when
+                     * notification service doesn't exist, which is the case in
+                     * external lib. We don't want notifications in external
+                     * lib at the moment because they use bootstrap toast and
+                     * we don't want to include boostrap in external lib.
+                     */
```

Comment:
Comment is good, I would consider at least adding a warn in the console to better allow debbuging of such issues.
And ideally it should display the error in another way, in the button or chat depending on when the issue happens.

Reply:
I prefer postponing the ideal display of error for when public livechat uses OWL.
Having notifications in the chat would require some IMP changes, which is not desirable before OWL conversion.

Therefore would limit with just console warning for the time being.